drivers/timer/ite_it8xxx2_timer.c

```diff
@@ -225,7 +225,15 @@ void sys_clock_set_timeout(int32_t ticks, bool idle)
 	IT8XXX2_EXT_CTRLX(EVENT_TIMER) &= ~IT8XXX2_EXT_ETXEN;
 
 	if (ticks == K_TICKS_FOREVER) {
-		/* Return since no future timer interrupts are required */
+		/*
+		 * If kernel doesn't have a timeout:
+		 * 1.CONFIG_SYSTEM_CLOCK_SLOPPY_IDLE = y (no future timer interrupts
+		 *   are expected), kernel pass K_TICKS_FOREVER (0xFFFF FFFF FFFF FFFF),
```

Comment:
Pedantry: the argument is an int32_t, so the kernel passes 0xffffffff or -1, not a 64 bit value.

Reply:
The parameter value is possibly the 64bit constant and therefore it should work consistently. Different compilers were treating the conversion from the possible int64_t to uint32_t with dropping the sign bit and others did not.  So made the comparison inconsistent.

Reply:
Which toolchain was having problems?  That sounds like something we'd want to tighten up in the timer headers if so.  Outside of warts like unsigned char, a -1 literal should remain a -1 literal through any sequence of casts;  but yes, I'm sure it's going to turn out to be Technically Undefined Behavior.